src/Serialization/Fields.jl

```diff
@@ -413,43 +426,85 @@ end
 ################################################################################
 # Field Embeddings
 
-@register_serialization_type Hecke.NumFieldEmbNfAbs  uses_id
+const FieldEmbeddingTypes = Union{Hecke.NumFieldEmbNfAbs, Hecke.NumFieldEmbNfRel, Hecke.NumFieldEmbNfAbsNS, Hecke.NumFieldEmbNfNS}
+
+@register_serialization_type Hecke.NumFieldEmbNfAbsNS uses_id
+@register_serialization_type Hecke.NumFieldEmbNfNS uses_id
+@register_serialization_type Hecke.NumFieldEmbNfAbs uses_id
+@register_serialization_type Hecke.NumFieldEmbNfRel uses_id
 
-function save_object(s::SerializerState, E::Hecke.NumFieldEmbNfAbs)
+function save_object(s::SerializerState, E::FieldEmbeddingTypes)
   K = number_field(E)
-  g = gen(K)
-  g_ball = E(g)
+  k = base_field(K)
 
   save_data_dict(s) do
     save_typed_object(s, K, :num_field)
-    save_typed_object(s, g_ball, :gen_ball)
+    if !(base_field(K) isa QQField)
+      save_typed_object(s, restrict(E, k), :base_field_emb)
+    end
+    if is_simple(K)
+      a = gen(K)
+      data = E(a)
+      if any(overlaps(data, e(a)) for e in complex_embeddings(K) if e != E && restrict(E, k) == restrict(e, k))
+        error("Internal error in internal serialization.")
+      end
+    else
+      a = gens(K)
+      data = E.(a)
+      if any(all(overlaps(t[1], t[2]) for t in zip(data, e.(a))) for e in complex_embeddings(K) if e != E && restrict(E, k) == restrict(e, k))
+        error("Internal error in internal serialization.")
+      end
+      data = tuple(data...)
+    end
+    save_typed_object(s, data, :data)
```

Comment:
so I am pretty sure you should be able to use 
save_object here, since you should be able to figure out what the type of data is based on what K is.

Then you can do something like

```
if is_simple(K)
  load_object(s, NotTuple, dict[:data]) ...
else
  load_object(s, Tuple, dict[:data], tuple_params)
end
```
for the load, might seem a bit more complicated, but it will cut down the size of the files, seeing as there are 92 files, it could make a big difference

Reply:
This has proven to be a bit more complex than expected - the `acb` in the `data` field needs an instantiated `AcbField` when loading this way. So the trade-off would be not as big and thus not rewarding enough for the changes needed.